### PR TITLE
Simplify JSON for submission for definition

### PR DESCRIPTION
**Pull request by birdbrained**

### CODE REVIEW

File: app/components/text-field/component.js [hunk 1]
```diff
@@ -3,15 +3,13 @@ import Ember from 'ember';
...
 
     textFieldValueObserver: Ember.observer('textFieldValue', function() {
-        const saveParameter = this.attrs.saveParameter;
-        const parameters = this.attrs.widget.value.parameters;
-        saveParameter(parameters.output, {
-            state: ['defined'],
-            value: this.get('textFieldValue'),
-        });
+        this.set('widget.parameters.value.value', this.get('textFieldValue'));
+        debugger;
```
Comment (cameronblandford) [inline app/components/text-field/component.js]: debugger!

File: app/components/text-field/template.hbs [hunk 2]
```diff
@@ -7,18 +7,16 @@
 
 </style>
 
-{{#if (not isOpen)}}
+{{#if (not editing)}}
 
-    {{valueName}}: {{textFieldValue}}
+    {{valueName}}: {{widget.parameter.value}}
```
Comment (cameronblandford) [inline app/components/text-field/template.hbs]: Re-examine this - should be `widget.parameter.value.value`

File: app/initializers/store.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import Ember from 'ember';
+import emberData from 'ember-data/-private';
```
Comment (cameronblandford) [inline app/initializers/store.js]: no

File: app/routes/collection/add.js [hunk 4]
```diff
@@ -5,21 +5,12 @@ export default Ember.Route.extend({
...
-        controller.set('sections', model.get('sections'));
-        controller.set('parameters', model.get('initialParameters'));
-
-        // Hydrate actions in preperation for engine ignition
-        const actions = model.get('actions').map(controller.hydrateAction.bind(controller));
-        controller.set('formActions', actions);
-
-        // Start the engine.
-        controller.updateState.call(controller, actions);
+        controller.set('workflow', model);
```
Comment (cameronblandford) [inline app/routes/collection/add.js]: :)

File: app/serializers/action.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+export default JSONSerializer.extend(DS.EmbeddedRecordsMixin, {
+
+    attrs: {
+        sections: {
+            embedded: 'always',
+        }
+    },
+
+    isSuccess(status) {
```
Comment (cameronblandford) [inline app/serializers/action.js]: review the necessity of these

File: app/serializers/parameter-mapping.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+    attrs: {
+        parameter: {
+            embedded: 'always',
+        }
+    },
+
+    isSuccess(status) {
+        if (status >= 200 && status < 300) return true;
+        if (status === 304) return true;
```
Comment (cameronblandford) [inline app/serializers/parameter-mapping.js]: remove?

File: app/serializers/workflow.js [hunk 7]
```diff
@@ -1,11 +1,183 @@
+import Ember from 'ember';
```
Comment (cameronblandford) [inline app/serializers/workflow.js]: Revisit this file, remove unnecessary parts

File: app/templates/collection/add.hbs [hunk 8]
```diff
@@ -1,46 +1,62 @@
...
+
+        --}}
+
+            <div class="section">
+                <div class="label">
+                    {{section.label}}
+                </div>
+                <div class="content">
+
+                    {{section.widgets}}
```
Comment (cameronblandford) [inline app/templates/collection/add.hbs]: is this for debugging?

File: app/templates/collection/add.hbs [hunk 9]
```diff
@@ -1,46 +1,62 @@
...
+                    {{#each section.widgets as |widget|}}
+                        {{widget.widgetType}}
```
Comment (cameronblandford) [inline app/templates/collection/add.hbs]: Remove as discussed

File: app/utils/hash-many.js [hunk 10]
```diff
@@ -0,0 +1,139 @@
+/**
```
Comment (cameronblandford) [inline app/utils/hash-many.js]: D-leet this

File: package.json [hunk 11]
```diff
@@ -62,14 +60,23 @@
...
+    "babel6-plugin-strip-heimdall": "^6.0.1",
+    "broccoli-debug": "^0.6.3",
+    "ember-cli": "2.14.1",
     "ember-cli-eslint": "^4.0.0",
+    "ember-cli-shims": "^1.1.0",
+    "ember-i18n": "^5.0.2",
+    "ember-lodash": "^4.17.5",
     "eslint-config-cos-ember": "^0.5.0",
     "moment": "^2.18.1",
+    "heimdall-query": "^0.0.7",
```
Comment (cameronblandford) [inline package.json]: remove

File: package.json [hunk 12]
```diff
@@ -62,14 +60,23 @@
...
+    "heimdalljs": "^0.2.5",
```
Comment (cameronblandford) [inline package.json]: remove

File: app/serializers/workflow.js (hunk 7 above)
Comment (birdbrained) [inline app/serializers/workflow.js]: 👍

**Pull request merged**